core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -582,6 +723,16 @@ public ShardRouting moveToUnassigned(ShardRouting shard, UnassignedInfo unassign
         return unassigned;
     }
 
+    private ShardRouting movePrimaryToUnassignedAndDemoteToReplica(ShardRouting shard, UnassignedInfo unassignedInfo) {
```

Comment:
Javadocs for this method would be great, even though it's private it would be nice to clarify the intent